routing/payment_lifecycle_test.go

```diff
@@ -627,6 +627,40 @@ func TestRouterPaymentStateMachine(t *testing.T) {
 			},
 			paymentErr: channeldb.FailureReasonPaymentDetails,
 		},
+		{
+			// A MP payment scenario when our path finding returns
+			// after we've just received a terminal failure, and
```

Comment:
Excellent bit of refactoring of the tests up to this point leading o the addition of this new reproduction test case!

Reply:
+1